File: src/main/java/io/github/starwishsama/namelessbot/session/Session.java
package io.github.starwishsama.namelessbot.session;

import lombok.Data;

import java.util.LinkedList;
import java.util.List;

/**
 * @author Nameless
 */
@Data
public class Session {
    private List<SessionUser> users = new LinkedList<>();
    private long groupId;

    public SessionUser getUserById(long id){
        if (!users.isEmpty()){
            for (SessionUser user : users){
                if (user.getUserId() == id){
                    return user;
                }
            }
        }
        return null;
    }
}

File: src/main/kotlin/io/github/starwishsama/nbot/sessions/Session.kt
package io.github.starwishsama.nbot.sessions

import io.github.starwishsama.nbot.commands.interfaces.UniversalCommand
import io.github.starwishsama.nbot.enums.SessionType
import java.util.*


/**
 * @author Nameless
 */
open class Session(var groupId: Long = 0, var type: SessionType, var command: UniversalCommand) {
    private var users: List<SessionUser> = LinkedList<SessionUser>()

    constructor(type: SessionType, command: UniversalCommand, id: Long) : this(0, type, command) {
        putUser(id)
    }

    fun putUser(id: Long) {
        users = users.plusElement(SessionUser(id))
    }

    fun getUserById(id: Long): SessionUser? {
        if (users.isNotEmpty()) {
            for (user in users) {
                if (user.userId == id) {
                    return user
                }
            }
        }
        return null
    }
}